terragraph_planner/los/base_los_validator.py

```diff
@@ -79,20 +104,20 @@ def _on_the_same_building(self, site1: LOSSite, site2: LOSSite) -> bool:
             and site1.building_id == site2.building_id
         )
 
-    def _los_out_of_distance_range(
-        self, site1: LOSSite, site2: LOSSite
+    def _los_out_of_distance_range(self, distance: float) -> bool:
+        return not self._min_los_distance <= distance <= self._max_los_distance
+
+    def _los_exceed_max_el_dev(
+        self,
+        distance: float,
+        altitude1: Optional[float],
+        altitude2: Optional[float],
     ) -> bool:
-        x_diff = site1.utm_x - site2.utm_x
-        y_diff = site1.utm_y - site2.utm_y
-        if site1.altitude is not None and site2.altitude is not None:
-            z_diff = site1.altitude - site2.altitude
-        else:
-            z_diff = 0
-        dist_sq = x_diff * x_diff + y_diff * y_diff + z_diff * z_diff
         return (
-            not self._min_los_distance_sq
-            <= dist_sq
-            <= self._max_los_distance_sq
+            altitude1 is not None
+            and altitude2 is not None
+            and math.degrees(math.asin(abs(altitude1 - altitude2) / distance))
+            > self._max_el_dev
```

Comment:
I think we should still return True for the non-dsm case, where the altitudes are both None. Otherwise, we will have no valid LOS when DSM is not provided.

```suggestion
            altitude1 is None
            or altitude2 is None
            or math.degrees(math.asin(abs(altitude1 - altitude2) / distance))
            > self._max_el_dev
```

Reply:
We return true when there's a violation and the if this is true, we return false in simple checks.

If altitude is None, this should return False so simple checks is ok.

Let me know if I have it backwards.

Reply:
My bad. You are right.